API/cron.ts

```diff
@@ -0,0 +1,144 @@
+import { VercelRequest, VercelResponse } from '@vercel/node';
+import { createClient } from '@supabase/supabase-js';
+import sgMail from '@sendgrid/mail';
+import { DateTime } from 'luxon';
+
+const supabaseUrl = process.env.NEXT_PUBLIC_SUPABASE_URL as string;
+const supabaseAnonKey = process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY as string;
+const sendgridApiKey = process.env.SENDGRID_API_KEY as string;
+const senderEmail = process.env.SENDGRID_SENDER_EMAIL as string;
+
+if (!sendgridApiKey) {
+    const error = new Error('SENDGRID_API_KEY was not provided as environment variable.') as Error & { code: number };
+    error.code = 401;
+    throw error;
+}
+
+if (!senderEmail) {
+    const error = new Error('SENDGRID_SENDER_EMAIL was not provided as environment variable.') as Error & { code: number };
+    error.code = 401;
+    throw error;
+}
+
+const supabase = createClient(supabaseUrl, supabaseAnonKey);
+sgMail.setApiKey(sendgridApiKey);
+
+
+
+export default async function handler(req: VercelRequest, res: VercelResponse) {
+    try {
+        const { data: notifications, error } = await supabase
```

Comment:
How does supabase deal with migrations to modify the existing tables? Haven't read much about it myself yet. 

For this project, we can afford to mess around a bit, but in general, it is important to ensure that the code is always compatible with the database structure, by applying DB migrations to make and undo changes as needed (sometimes this is destructive to data, so better be careful and have some kind of system for backups), such as modifying the nullability of certain columns, adding new ones, etc.